Conditionally pin PyYAML back on python 3.4

tfoote opened the pull request:
The latest version is no longer backwards compatible.

This is following the pattern we've used elsewhere such as: https://github.com/ros-infrastructure/rosdep/pull/739

tfoote committed:
Conditionally pin PyYAML back on python 3.4

The latest version is no longer backwards compatible.

wjwwood commented:
It lgtm, but does fix CI apparently. We could also consider dropping testing for 3.4? Is it still used in one of our LTS releases or something?

tfoote committed:
remove the update flag

tox requires pyyaml and it's trying to upgrade it to an incompatible version with the `-U` flag

Signed-off-by: Tully Foote <tfoote@osrfoundation.org>

tfoote commented:
Jessie is almost EOL but is still using 3.4 https://packages.debian.org/jessie/python3

tfoote commented:
It looks like the pyyaml version needs to be pinned inside the tox environment as well as in the main travis invocation

cottsay commented:
@tfoote - any chance we can get this fixed up so that #163 can be merged?

tfoote committed:
disable 3.4 within tox

tfoote commented:
I tried just turning off 3.4 inside of the tox.ini Interestingly both it and travis appear to have a matrix. I think we might be running venvs inside venvs. I don't have a better solution than I think that I"m not sure what tox is doing for us besides running the tests natively in travis.

tfoote committed:
try running setup.py test direction

dirk-thomas commented:
Closing since the repository is about to be archived: see #166.
